src/accel/LocalTransporter.cc

```diff
@@ -37,17 +39,34 @@ LocalTransporter::LocalTransporter(SPConstOptions opts, SPConstParams params)
     }
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Set the event ID at the start of an event.
+ */
+void LocalTransporter::SetEventId(int id)
+{
+    CELER_EXPECT(id >= 0);
+    event_id_ = EventId(id);
+}
+
 //---------------------------------------------------------------------------//
 /*!
  * Convert a Geant4 track to a Celeritas primary and add to buffer.
  */
-void LocalTransporter::add(const G4Track& g4track)
+bool LocalTransporter::TryOffload(const G4Track& g4track)
```

Comment:
Probably better to decouple the offload triggering condition from the transporter, i.e., move TryOffload to UserAction (TrackingAction in this case) as the user will decide an offloading condition based on particle types, detector regions and etc.

Reply:
Good point. Should we add an assertion that we know the particle type? We're going to have to be careful with the "triggering" condition since we can't move tracks back to the host...